### PR TITLE
Changes to urfave CLI, implements version command and one-page doc

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This changes to urfave CLI, which is easier to work with than Cobra.

Pros of both:
* long source history, large amount of stars, reasonably frequent releases
* support the platforms we need (macOS, Windows, linux)
* support things we need in the future like auto-completion #115 

Pros of urfave:
* significantly less indirect dependencies (283 less lines in go.sum)
* supports ENV variables without introducing another dependency
* cleaner support for go context (haven't used it yet)
* less awkward function names and behaviors (ex. `Before` instead of `PersistentPreRunE`)
* simpler help output and markdown (ex single page vs multi-page)
* slightly smaller binary (They even [measure](https://github.com/urfave/cli/blob/45952a7d1b0e956db2099493fe1a42837ff5bd0d/internal/build/build.go#L186-L202) it!)

Cons of urfave
* doesn't yet support [argument metadata](https://github.com/urfave/cli/issues/1074)
  * We only have two args, the `<reference>` and then anything passed to envoy `-- .*`, so this isn't a biggie

https://github.com/urfave/cli/blob/master/docs/v2/manual.md

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,120 +1,25 @@
-cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (codefromthecrypt) [inline go.sum:1]: 283 lines gone from the go.sum file

File: pkg/cmd/testdata/getenvoy.md [hunk 2]
```diff
@@ -1,27 +1,51 @@
-+++
-title = "getenvoy"
-type = "reference"
-parent = "root"
-command = "getenvoy"
-+++
-## getenvoy
+% getenvoy 8
```
Comment (codefromthecrypt) [inline pkg/cmd/testdata/getenvoy.md:1]: need to polish the format template, but in general it is much better IMHO to do single-page markdown for such a small utility.

File: pkg/cmd/run.go [hunk 3]
```diff
@@ -31,34 +31,51 @@ import (
...
-			r.Err = c.ErrOrStderr()
+
+			r.Out = c.App.Writer
+			r.Err = c.App.ErrWriter
 
 			debug.EnableAll(r)
 
-			envoyArgs := args[1:]
-			return r.Run(c.Context(), envoyArgs)
+			envoyArgs := findEnvoyArgs(args)
```
Comment (llinder) [inline pkg/cmd/run.go:58]: I like the clarity of this 👍

**Comment by codefromthecrypt** [comment]
I will try to finish this tomorrow morning asia time

**Comment by codefromthecrypt** [comment]
once this is merged, I will help simplify the getenvoy-ui usage of "getenvoy doc" to emit the now much simpler single page. This should suffice, but if we later feel we want to edit it, we can override the template like so https://github.com/google/skia-buildbot/blob/4c87209ff9378a9befc65d5f93c4040d4fdd5922/go/urfavecli/urfavecli.go#L34 eventually, we should make markdown completely driven by the UI project instead of a hidden command as mentioned in #205

File: pkg/cmd/testdata/getenvoy.md (hunk 2 above)
Comment (mathetake) [reply pkg/cmd/testdata/getenvoy.md:1]: I guess this breaks hugo builds

Comment (mathetake) [reply pkg/cmd/testdata/getenvoy.md:1]: we need +++ block

Comment (codefromthecrypt) [reply pkg/cmd/testdata/getenvoy.md:1]: magic info :) can't wait until we have things in the same repo

Comment (codefromthecrypt) [reply pkg/cmd/testdata/getenvoy.md:1]: agree even latest hugo can't handle this. will make a change

Comment (codefromthecrypt) [reply pkg/cmd/testdata/getenvoy.md:1]: this isn't really hugo related as normal markdown breaks rendering. It isn't broken, per-se, just it adds a useless line. I raised https://github.com/urfave/cli/issues/1275

**Comment by codefromthecrypt** [comment]
I have a WIP branch for the website that correctly renders now https://github.com/tetratelabs/getenvoy.io/compare/new-usage

Screen Shot 2021-05-21 at 9 28 09 AM

File: pkg/cmd/app_test.go [hunk 4]
```diff
@@ -178,25 +178,23 @@ func requireSetenv(t *testing.T, key, value string) func() {
...
 	c, stdout, stderr := newApp(o)
-	c.AddCommand(&cobra.Command{
-		Use: "test",
-		RunE: func(cmd *cobra.Command, _ []string) error {
-			return nil
-		},
-	})
-	c.SetArgs(append(args, "test"))
-	err := c.Execute()
+	c.Commands = append(c.Commands, &cli.Command{Name: "test", Action: func(context *cli.Context) error {
```
Comment (mathetake) [inline pkg/cmd/app_test.go:193]: nit
```suggestion
	c.Commands = append(c.Commands, &cli.Command{Name: "test", Action: func(cli.Context) error {
```

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the review @llinder and @mathetake!

**Pull request merged**